File: examples/example_node.rs
use std::thread;
use std::time::Duration;
use rand::Rng;
use space_net::node::*;
use space_net::subscriber::NodeSubscriber;
use zenoh::subscriber::Subscriber;

fn main() {
    let mut rng = rand::thread_rng();
    let point = (rng.gen_range(1.0..=99.0), rng.gen_range(1.0..=99.0));
    let mut node = Node::new("network_1");
    node.leave_on_pressed('q');
    node.join(point);
    println!("Node online..... {:?}", node.get_zid());


    loop {
        if node.get_status() == NodeStatus::Offline {
            break;
        }
    }
}

File: examples/ExampleNode.java
import com.example.Node;

public class ExampleNode {
    public static void main(String[] args) {
        Node node = new Node("network_1");
         String zid = node.getZid();
         System.out.println("Node zid: " + zid);
         node.leaveOnKey('q');
         node.join(69.0,69.0);

         while (true) {

         }
    }
}
